spec/lib/miq_automation_engine/service_methods/miq_ae_service_vm_spec.rb

```diff
@@ -149,17 +149,61 @@ def invoke_ae
     end
 
     it "#retires_on - today" do
-      service_vm.retires_on = Date.today
-      vm.reload
+      vm.update_attributes(:retirement_last_warn => Date.today)
+      service_vm.retires_on = Time.zone.today
 
+      vm.reload
+      expect(vm.retirement_last_warn).to be_nil
       expect(vm.retirement_due?).to be_truthy
     end
 
     it "#retires_on - tomorrow" do
-      service_vm.retires_on = Date.today + 1
+      vm.update_attributes(
```

Comment:
@tinaafitz In every test that you're doing `vm.update_attributes` as part of set-up code, you're causing an extra write to the database that slow down tests. `vm` calls the memoized FactoryGirl call at the top of this file and `update_attributes` updates the record again.

You can avoid this by just overriding the helper to begin with, within a context:

```ruby
context "tomorrow" do
  let(:vm) { FactoryGirl.create(:vm_vmware, :retired => true }  #etc etc etc
  it "<description of what it should be>" do
    # ...
  end
end
```

This is only a single write for this test.